tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/series/FileSeriesReader.java

```diff
@@ -16,95 +16,40 @@
  */
-
 package org.apache.iotdb.tsfile.read.reader.series;
 
 import java.io.IOException;
 import java.util.List;
-import org.apache.iotdb.tsfile.file.header.PageHeader;
 import org.apache.iotdb.tsfile.file.metadata.ChunkMetaData;
-import org.apache.iotdb.tsfile.read.common.BatchData;
+import org.apache.iotdb.tsfile.read.common.Chunk;
 import org.apache.iotdb.tsfile.read.controller.IChunkLoader;
+import org.apache.iotdb.tsfile.read.filter.basic.Filter;
 import org.apache.iotdb.tsfile.read.reader.chunk.ChunkReader;
 
 /**
- * Series reader is used to query one series of one tsfile.
+ * Series reader is used to query one series of one TsFile,
+ * and this reader has a filter operating on the same series.
  */
-public abstract class FileSeriesReader {
+public class FileSeriesReader extends AbstractFileSeriesReader {
 
-  protected IChunkLoader chunkLoader;
-  protected List<ChunkMetaData> chunkMetaDataList;
-  protected ChunkReader chunkReader;
-  private int chunkToRead;
+  public FileSeriesReader(IChunkLoader chunkLoader,
+      List<ChunkMetaData> chunkMetaDataList, Filter filter) {
+    super(chunkLoader, chunkMetaDataList, filter);
+  }
 
-  private BatchData data;
 
-  /**
-   * constructor of FileSeriesReader.
-   */
-  public FileSeriesReader(IChunkLoader chunkLoader, List<ChunkMetaData> chunkMetaDataList) {
-    this.chunkLoader = chunkLoader;
-    this.chunkMetaDataList = chunkMetaDataList;
-    this.chunkToRead = 0;
+  @Override
+  protected void initChunkReader(ChunkMetaData chunkMetaData) throws IOException {
+    Chunk chunk = chunkLoader.getChunk(chunkMetaData);
+    this.chunkReader = new ChunkReader(chunk, filter);
   }
 
-  /**
-   * check if current chunk has next batch data.
-   *
-   * @return True if current chunk has next batch data
-   */
-  public boolean hasNextBatch() throws IOException {
-
-    // current chunk has additional batch
-    if (chunkReader != null && chunkReader.hasNextBatch()) {
+  @Override
+  protected boolean chunkSatisfied(ChunkMetaData chunkMetaData) {
+    if (filter == null ) {
       return true;
     }
-
-    // current chunk does not have additional batch, init new chunk reader
-    while (chunkToRead < chunkMetaDataList.size()) {
-
-      ChunkMetaData chunkMetaData = nextChunkMeta();
-      if (chunkSatisfied(chunkMetaData)) {
-        // chunk metadata satisfy the condition
-        initChunkReader(chunkMetaData);
-
-        if (chunkReader.hasNextBatch()) {
-          return true;
-        }
-      }
-    }
-    return false;
-  }
-
-  /**
-   * get next batch data.
-   */
-  public BatchData nextBatch() throws IOException {
-    data = chunkReader.nextBatch();
-    return data;
-  }
-
-  public BatchData currentBatch() {
-    return data;
-  }
-
-  public PageHeader nextPageHeader() throws IOException {
-    return chunkReader.nextPageHeader();
-  }
-
-  public void skipPageData() {
-    chunkReader.skipPageData();
+    return filter.satisfy(chunkMetaData.getStatistics());
```

Comment:
Method `chunkSatisfied` could changed to:
```suggestion
    return filter == null || filter.satisfy(chunkMetaData.getStatistics());
```

Reply:
fixed